cmd/monitoring/monitoring.go

```diff
@@ -83,18 +83,20 @@ func (m *monitor) listResourceGroupMonitoringHostnames(ctx context.Context, subs
 		}
 	}
 	// get all external IP's used by VMSS
+	m.log.Debug("waiting for ss-masters ip addresses")
 	hostnames = []string{}
 	for iter, err := m.pipcli.ListVirtualMachineScaleSetPublicIPAddressesComplete(ctx, resourceGroup, "ss-master"); iter.NotDone(); err = iter.Next() {
 		if err != nil {
-			return nil, err
+			time.Sleep(5 * time.Second)
```

Comment:
do we not lose IP addresses if we do this?

Reply:
We might. But we will see less IP in the metrics. 
If this returns err - we have a bigger problem. Not sure we want to try catching ARM errors here as it should be part of plugin code

Reply:
we're not aligned.  these days the fake rp writes _data/containerservice.yaml before (and after) the cluster is created so you *will* end up here before the IPs exist.  If you can, poll locally to be sure the cluster is created (can you read the provisioning state perhaps?) then fetch all the IPs.  Adding `time.Sleep(5 * time.Second)` without a loop basically achieves nothing.

Reply:
Got you. Bringing back loop. provisioning state means I still call "something" each 5 sec to get state, and after that, I call IP addresses and LB. So this would mean one more client to configure. I think this is simpler.

Reply:
There is issue with it now: https://github.com/openshift/openshift-azure/issues/1416 Fixing in separate PR.